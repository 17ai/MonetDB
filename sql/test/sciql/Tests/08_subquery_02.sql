# For now, disable mitosis for all SciQL GROUP BY queries, because mitosis
# changes the HEAD of a BAT, which in turn breaks the array AGGR
# implementations.  Fixing this might require extension at the MAL or even GDK
# level, postponed ...

set optimizer='no_mitosis_pipe';

CREATE ARRAY ary(x INT DIMENSION[4], y INT DIMENSION[-5], v FLOAT DEFAULT 3.7);

WITH a AS (SELECT * FROM ary[1:3]) SELECT x, y, SUM(v), AVG(v) FROM a GROUP BY a[x-1:x+1][y+1:y-1];

WITH a AS (SELECT * FROM ary[1:3]), b AS(SELECT * FROM ary[*][-1:-3]) SELECT x, y, SUM(v), AVG(v) FROM b GROUP BY b[x-1:x+1][y+1:y-1];

WITH a AS (SELECT * FROM ary[1:3]), b AS(SELECT * FROM ary[*][-1:-3]) SELECT x, y, SUM(v), AVG(v) FROM a GROUP BY b[x-1:x+1][y+1:y-1];
WITH a AS (SELECT * FROM ary[1:3]), b AS(SELECT * FROM ary[*][-1:-3]) SELECT x, y, SUM(v), AVG(v) FROM b GROUP BY a[x-1:x+1][y+1:y-1];

DROP ARRAY ary;

