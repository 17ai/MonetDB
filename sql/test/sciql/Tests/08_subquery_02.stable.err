stderr of test '08_subquery_02` in directory 'sql/test/sciql` itself:


# 15:01:40 >  
# 15:01:40 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "gdk_dbfarm=/export/scratch2/zhang/monet-install/sciql/debug/var/MonetDB" "--set" "mapi_open=true" "--set" "mapi_port=32229" "--set" "monet_prompt=" "--trace" "--forcemito" "--set" "mal_listing=2" "--dbname=mTests_test_sciql" "--set" "mal_listing=0"
# 15:01:40 >  

# builtin opt 	gdk_dbname = demo
# builtin opt 	gdk_dbfarm = /export/scratch2/zhang/monet-install/sciql/debug/var/monetdb5/dbfarm
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_alloc_map = no
# builtin opt 	gdk_vmtrim = yes
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	gdk_dbfarm = /export/scratch2/zhang/monet-install/sciql/debug/var/MonetDB
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 32229
# cmdline opt 	monet_prompt = 
# cmdline opt 	mal_listing = 2
# cmdline opt 	gdk_dbname = mTests_test_sciql
# cmdline opt 	mal_listing = 0

# 15:01:40 >  
# 15:01:40 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=riga" "--port=32229"
# 15:01:40 >  

MAPI  = (monetdb) /var/tmp/mtest-29912/.s.monetdb.33795
QUERY = WITH a AS (SELECT * FROM ary[1:3]), b AS(SELECT * FROM ary[*][-1:-3]) SELECT x, y, SUM(v), AVG(v) FROM a GROUP BY b[x-1:x+1][y+1:y-1];
ERROR = !SELECT: no such array 'sys.b'
MAPI  = (monetdb) /var/tmp/mtest-29912/.s.monetdb.33795
QUERY = WITH a AS (SELECT * FROM ary[1:3]), b AS(SELECT * FROM ary[*][-1:-3]) SELECT x, y, SUM(v), AVG(v) FROM b GROUP BY a[x-1:x+1][y+1:y-1];
ERROR = !SELECT: no such array 'sys.a'

# 15:01:40 >  
# 15:01:40 >  "Done."
# 15:01:40 >  

