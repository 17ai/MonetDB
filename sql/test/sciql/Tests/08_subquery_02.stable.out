stdout of test '08_subquery_02` in directory 'test/sciql` itself:


# 15:01:53 >  
# 15:01:53 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "gdk_dbfarm=/export/scratch2/zhang/monet-install/sciql/debug/var/MonetDB" "--set" "mapi_open=true" "--set" "mapi_port=30528" "--set" "monet_prompt=" "--trace" "--forcemito" "--set" "mal_listing=2" "--dbname=mTests_test_sciql" "--set" "mal_listing=0"
# 15:01:53 >  

# MonetDB 5 server v11.10.0
# This is an unreleased version
# Serving database 'mTests_test_sciql', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 15.629 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2012 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://riga.ins.cwi.nl:30528/
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded

Ready.
# SQL catalog created, loading sql scripts once
# loading sql script: 09_like.sql
# loading sql script: 10_math.sql
# loading sql script: 11_times.sql
# loading sql script: 12_url.sql
# loading sql script: 13_date.sql
# loading sql script: 14_inet.sql
# loading sql script: 15_history.sql
# loading sql script: 16_tracelog.sql
# loading sql script: 17_compress.sql
# loading sql script: 18_dictionary.sql
# loading sql script: 19_cluster.sql
# loading sql script: 20_vacuum.sql
# loading sql script: 21_dependency_functions.sql
# loading sql script: 22_clients.sql
# loading sql script: 23_skyserver.sql
# loading sql script: 24_zorder.sql
# loading sql script: 25_debug.sql
# loading sql script: 29_array.sql
# loading sql script: 39_analytics.sql
# loading sql script: 40_geom.sql
# loading sql script: 80_udf.sql
# loading sql script: 99_system.sql

# 15:01:53 >  
# 15:01:53 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=riga" "--port=30528"
# 15:01:53 >  

#set optimizer='no_mitosis_pipe';
#CREATE ARRAY ary(x INT DIMENSION[4], y INT DIMENSION[-5], v FLOAT DEFAULT 3.7);
#WITH a AS (SELECT * FROM ary[1:3]) SELECT x, y, SUM(v), AVG(v) FROM a GROUP BY a[x-1:x+1][y+1:y-1];
% sys.a,	sys.a,	sys.,	sys. # table_name
% x,	y,	L2,	L3 # name
% int,	int,	double,	double # type
% 1,	2,	24,	24 # length
[ 1,	0,	3.7,	3.7	]
[ 1,	-1,	7.4,	3.7	]
[ 1,	-2,	7.4,	3.7	]
[ 1,	-3,	7.4,	3.7	]
[ 1,	-4,	7.4,	3.7	]
[ 2,	0,	7.4,	3.7	]
[ 2,	-1,	14.8,	3.7	]
[ 2,	-2,	14.8,	3.7	]
[ 2,	-3,	14.8,	3.7	]
[ 2,	-4,	14.8,	3.7	]
#WITH a AS (SELECT * FROM ary[1:3]), b AS(SELECT * FROM ary[*][-1:-3]) SELECT x, y, SUM(v), AVG(v) FROM b GROUP BY b[x-1:x+1][y+1:y-1];
% sys.b,	sys.b,	sys.,	sys. # table_name
% x,	y,	L3,	L4 # name
% int,	int,	double,	double # type
% 1,	2,	24,	24 # length
[ 0,	-1,	3.7,	3.7	]
[ 0,	-2,	7.4,	3.7	]
[ 1,	-1,	7.4,	3.7	]
[ 1,	-2,	14.8,	3.7	]
[ 2,	-1,	7.4,	3.7	]
[ 2,	-2,	14.8,	3.7	]
[ 3,	-1,	7.4,	3.7	]
[ 3,	-2,	14.8,	3.7	]
#DROP ARRAY ary;

# 15:01:53 >  
# 15:01:53 >  "Done."
# 15:01:53 >  

